src/front/zsharpcurly/mod.rs

```diff
@@ -1814,22 +1876,25 @@ impl<'ast> ZGen<'ast> {
             );
         }
 
-        if let Some(ast::ArrayParamMetadata::Transcript(_)) = &c.array_metadata {
-            if !value.type_().is_array() {
-                self.err(format!("Non-array transcript {}", &c.id.value), &c.span);
-            }
-            self.mark_array_as_transcript(&c.id.value, value.clone());
-        }
+        // if let Some(ast::ArrayParamMetadata::Transcript(_)) = &c.array_metadata {
```

Comment:
Again, I suggest throwing an error.

Reply:
I removed the commented-out code from here. Basically, the new AST does not contain ArrayParamMetadata